packages/wdio-utils/src/envDetector.ts

```diff
@@ -119,7 +125,15 @@ function isAndroid (caps) {
  * @param  {object}  capabilities session capabilities
  * @return {Boolean}              true if session is running on Sauce with extended debugging enabled
  */
-function isSauce (caps) {
+function isSauce (capabilities?: WebDriver.DesiredCapabilities | WebDriver.W3CCapabilities) {
+    if (!capabilities) {
+        return false
+    }
+
+    const caps: WebDriver.DesiredCapabilities = (capabilities as WebDriver.W3CCapabilities).alwaysMatch
```

Comment:
Is this some kind of bugfix in this PR?

Reply:
Yeah, detecting the Sauce background would not have worked using pure WebDriver capabilities with the `webdriver` package